Libraries/dotNetRDF/Core/GraphDiffReportExtensions.cs

```diff
@@ -0,0 +1,130 @@
+        /// <returns>A <see cref="ModifyCommand">SPARQL Update INSERT/DELETE command</see> that represents the <see cref="GraphDiffReport">diff</see></returns>
+        public static ModifyCommand AsUpdate(this GraphDiffReport diff, Uri graphUri = null, INamespaceMapper prefixes = null)
+        {
+            var delete = new GraphPatternBuilder();
+            var insert = new GraphPatternBuilder();
+            var where = new GraphPatternBuilder();
+
+            // Removed ground triples are added as is to both delete and where clauses
+            foreach (var t in diff.RemovedTriples)
+            {
+                delete.AddTriplePattern(t);
+                where.AddTriplePattern(t);
+            }
+
+            foreach (var g in diff.RemovedMSGs)
+            {
+                // Blank nodes in removed non-ground triples are converted to variables and added to both delete and where clauses
+                foreach (var t in g.Triples)
+                {
+                    delete.AddVariablePattern(t);
+                    where.AddVariablePattern(t);
+                }
+
+                // An ISBLANK filter is added for each blank node in removed non-ground triples
+                foreach (var n in g.BlankNodes())
+                {
+                    where.AddBlankNodeFilter(n);
+                }
+            }
+
+            // Added triples (ground or not) are added as is to the insert clause
+            foreach (var t in diff.AllAddedTriples())
+            {
+                insert.AddTriplePattern(t);
+            }
+
+            return new ModifyCommand(
+                delete.BuildGraphPattern(prefixes),
+                insert.BuildGraphPattern(prefixes),
+                where.BuildGraphPattern(prefixes),
+                graphUri);
+        }
+
+        private static void AddTriplePattern(this IGraphPatternBuilder builder, Triple t)
+        {
+            builder
+                .Where(tripleBuilder =>
+                    tripleBuilder
+                        .Subject(t.Subject)
+                        .PredicateUri(t.Predicate as IUriNode)
+                        .Object(t.Object));
+        }
+
+        private static void AddVariablePattern(this IGraphPatternBuilder builder, Triple t)
+        {
+            builder
+                .Where(tripleBuilder =>
+                    tripleBuilder
+                        .Subject(t.Subject.AsVariable())
+                        .PredicateUri(t.Predicate as IUriNode)
+                        .Object(t.Object.AsVariable()));
+        }
+
+        private static void AddBlankNodeFilter(this IGraphPatternBuilder builder, IBlankNode n)
+        {
+            builder.Filter(node =>
+                node.IsBlank(n.InternalID));
+        }
+
+        private static INode AsVariable(this INode n)
+        {
+            return n is IBlankNode blank ? new NodeFactory().CreateVariableNode(blank.InternalID) : n;
```

Comment:
I'd agree with @tpluscode. Although you aren't using the blank node mapper in the NodeFactory instance right now, it would be good to be using a single factory for the whole of the command generation process just to avoid a bunch of short term objects on the heap.